2019-03-Autumn/1_linked_lists/README.md

```diff
@@ -102,4 +135,196 @@ TODO :bug:
 
 TODO :bug:
 
-
+### 3. Sum Lists
+
+Source: CTCI 2.5
+
+#### Solution for Reversed List (First Problem Statement)
+
+Because the order in which we add the digits is the same order in which 
+they are presented, we iterate through the lists in lockstep and add 
+the `sum % 10` of the respective list nodes to a new list we call `result`. 
+This approach requires checking whether the either of the current nodes 
+are `null`, and when both are `null`, the loop terminates.
+
+The code is given as:
+
+```c++
+Node* sumReversedNums(Node* n1, Node* n2) {
+    // build list for resulting sum
+    Node* result = nullptr;
+    Node* end = result;
+    int carry = 0;
+    // iterate through argument lists
+    while (n1 != nullptr || n2 != nullptr) {
+        int sum = carry;
+        if (n1 != nullptr) {
+            sum += n1->data;
+            n1 = n1->next;
+        }
+        if (n2 != nullptr) {
+            sum += n2->data;
+            n2 = n2->next;
+        }
+        carry = (sum > 9 ? 1 : 0);
+        // add result to result list
```

Comment:
`carry = sum / 10;`

Reply:
I'm leaving it as a ternary operator, since the intent is more explicit here and doesn't rely on language-specific handling of floating-point vs integer division. (Yes, I know that you can specify integer division in Python by doing `//`, but I'm sticking to my ternary ops because I like them :stuck_out_tongue: )